### ISSUE TITLE
PHAR build broken ?

### ISSUE THREAD

**Issue by marmotz** [atoum/atoum#383]
The last PHAR archive seems to be old:

```
$ php mageekguy.atoum.phar -version
atoum version nightly-2416-201402121146 by Frédéric Hardy

$ php -d phar.readonly=0 vendor/mageekguy.atoum.phar -update
Checking if a new version is available... Done !
There is no new version available !
```

**Comment by Grummfy** [comment]
sinon, pour faire simple et résoudre le problème, je propose ceci : 
* changer le code lié à cette vérification afin de demander plutôt  la dernière version a github (ou reprendre celle-ci depuis github depuis le serveur de atoum)
* en cas de mise à jour, prendre le phar sur github
* packager dans un phar
* updater manuellement le phar présent a l'endroit du check.
* et voilà

**Comment by Grummfy** [comment]
So, some stuff from irc :
to build the nightly, mageekguy use the following steps :
using this phar : http://medias.atoum.org/mageekguy.atoum.phar
> /usr/local/bin/php -f /usr/local/www/downloads.atoum.org/mageekguy.atoum.phar -- -u builder -c /usr/local/www/downloads.atoum.org/configurations/builder.php -r https://github.com/atoum/atoum/trunk -p /usr/local/bin/php
config file is  :
> https://gist.github.com/mageekguy/86ed08ff5289918f82ad

error : 
> Error: Error: file_put_contents(/usr/local/www/downloads.atoum.org/configurations/../tmp/builder/src/.): failed to open stream: Is a directory


I try but failed to build locally with this command or (on my computer)
> php atoum/scripts/builder.php -c config-build.php -r https://github.com/atoum/atoum/trunk  -p /usr/bin/php

But the command to build one, unique phar is ok : 
> php -dphar.readonly=Off atoum/scripts/phar/generator.php -d tmp/

Hope it will help

**Comment by Hywan** [comment]
Nobody really answered to my comment: https://github.com/atoum/atoum/issues/383#issuecomment-66422337 :-p.

**Comment by jubianchi** [comment]
@Hywan the documentation is more like rolling-release ATM.

Before planning any tag on it we should focus on making it exhaustive on existing (yet not documented) features.

**Comment by Hywan** [comment]
@jubianchi What's the link with the documentation?

**Comment by jubianchi** [comment]
@Hywan ho sorry I read the wrong comment :/ But **you** talked about the doc :p

https://github.com/atoum/atoum/issues/383#issuecomment-66444036

**Comment by jubianchi** [comment]
@Hywan 

> Can't it be an HTTP redirect to Github?

No because the actual Phar is a nightly not a stable release.

**Comment by Grummfy** [comment]
yep. but we should evolve and the command could be update to stable or unstable (nightly) stuff.

**Comment by jubianchi** [comment]
@Grummfy stable phars are already released by @Hywan and @mikaelrandy which each atoum release. 

The nightly phar still makes sense and we have to fix it... I think it should not be too hard... perhaps just updating the phar used to build other phars will do it ;)

**Comment by mageekguy** [comment]
@Grummfy you should download http://medias.atoum.org/mageekguy.atoum.phar and https://gist.github.com/mageekguy/86ed08ff5289918f82ad and update the phar path and the configuration file path accordingly in `/usr/local/bin/php -f path/to/mageekguy.atoum.phar -- -u builder -c path/to/configuration/file -r https://github.com/atoum/atoum/trunk -p /usr/local/bin/php` (ie. replace `path/to/mageekguy.atoum.phar` and `path/to/configuration/file` with your local path respectively).

**Comment by Grummfy** [comment]
The purpose of this nightly builded phar is to have both a phar to dl and to make update from a phar locally installed.
What I think that could be do (cf https://github.com/atoum/atoum/issues/383#issuecomment-72618744 ) is a way to update local phar with the n ightly or the stable release. But it's more another issue => https://github.com/atoum/atoum/issues/428

Well @mageekguy I try and failed : sometimes sqlite error sometime other. With the given phar I have another error (but not with a new atoum phar)

**Comment by Grummfy** [comment]
So nobody want to fix it?

**Comment by Hywan** [comment]
Should we update to stable versions only and fetch the PHAR archive from Github? If yes, this is not really difficult to patch.

**Comment by Grummfy** [comment]
So, can we do something about this one?
Perhaps a redirection for the last stable version on one side (so update will use last stable one) and use travis to build nightly?

**Comment by jubianchi** [comment]
I think providing two update channels (stable and unstable) is something we should do. Users will be able to choose from which channel they want to update.

Using Github releases for stable channel seems ok. The only problem I see is that updates will be broken each time Github is down (which happens not so often). The other downside of this is we will have github specific code in the core of atoum which is unlikely.

To solve this problem, we could use something like a proxy : atoum will fetch an URL on its own domain (i.e http://update.atoum.org/stable). Something like what @Hywan did for Hoa resources.

Next thing is the unstable channel : using travis to build and deploy nightlies could also be a good thing : we will only have to figure out the right configuration to tell travis to build our phar and push it somewhere. I don' treally know what is supported by travis (where and how it can upload things).

But I'm :+1: with what you propose here: everything (more or less) will live in the repository so anyone will be able to fix it when something is not working as expected.

**Comment by Grummfy** [comment]
1. We will have http://update.atoum.org/stable for stable release that will use the api of github or whatever to have all stable realeased there
2. Have http://update.atoum.org/unstable with all nightly (we can imagine that travis ping a builder on the website of atoum, it can be as simple as that, we just need to figure a way to securise it)
3. Once this is in place, we need to update the upgrade process of atoum to use the good version of atoum.

Seems to be a plan. I create tree issue link to this one?

**Comment by Grummfy** [comment]
travis build upload on finished : http://docs.travis-ci.com/user/deployment/custom/ or with 
http://docs.travis-ci.com/user/deployment/script/

**Comment by Grummfy** [comment]
Hello,
on many tutorial, including travis the way to downlaod to use atoum is by the phar.
So here is what I think that can help : 
* redirect the phar download to the last version of the phar on github (manually if we need) => this need to be done, I think, and don't take so much works!
* rethink if it's very usefull to have a nightly build of the phar
* if it's usefull find a way to build it correctly (I can check)

**Comment by Hywan** [comment]
https://github.com/atoum/atoum/issues/383#issuecomment-77824146 :-)
We must use PHAR attached to Github releases.

**Comment by Grummfy** [comment]
oki.
I will create a splitted issue for the this task.

For the nightly, I thnik we should wait at least 2 other comment before removing them.

**Comment by Hywan** [comment]
#518 has been merged.

**Comment by Grummfy** [comment]
For me this is not closed. The people who follow our documentation and the documentation of travis need to have the url http://downloads.atoum.org/nightly/mageekguy.atoum.phar getting the *LAST* version of atoum (or at least the last stable version).

Until this is correct for be there is still an issue ... (#404)

**Comment by Hywan** [comment]
@mageekguy Can we update the link on downloads.atoum.org please?

**Comment by Grummfy** [comment]
up.... Honestly this is a very big problem we need to have a solution

**Comment by Hywan** [comment]
We must redirect `downloads.atoum.org/……` to Github. That's all.

**Comment by Grummfy** [comment]
So what's the missing point ? How to accelerate it? asking @mageekguy  ?
